src/test/java/com/elasticbox/jenkins/tests/PullRequestManagerTest.java

```diff
@@ -0,0 +1,279 @@
+        Thread.sleep(5000);
+        List<GHHook> hooks = gitHubRepo.getHooks();
+        GHHook webhook = null;
+        for (GHHook hook : hooks) {
+            if ("web".equals(hook.getName()) && webhookUrl.equals(hook.getConfig().get("url"))) {
+                webhook = hook;
+                break;
+            }
+        }
+        Assert.assertNotNull(MessageFormat.format("Webhook {0} is not created for repository {1}", webhookUrl, gitHubRepo.getHtmlUrl()), webhook);
+
+        LOGGER.fine("launching one pull request that triggers job 1\n");
+        pullRequest.open();
+        Thread.sleep(10000);
+
+        LOGGER.fine("Check that the job 0 is triggered");
+        new Condition() {
+            public boolean satisfied() {
+                return project.getLastBuild() != null;
+            }
+        }.waitUntilSatisfied(NEXTBUILD_TIMEOUT_LONG);
+
+        Assert.assertNotNull(MessageFormat.format("Build is not triggered on opening of pull request {0} after {1} minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), TimeUnit.SECONDS.toMinutes(NEXTBUILD_TIMEOUT_LONG)),
+                project.getLastBuild() );
+
+        FreeStyleBuild projectLastBuild1_0 = project.getLastBuild();
+        LOGGER.fine("The job 0 have triggered build with number " + projectLastBuild1_0.number);
+
+
+        LOGGER.fine("Creating new job 2 listening trigger from the same repository...\n");
+        FreeStyleProject project2 = createJenkinsJob2();
+
+        testAddingProjectToGHRepository(project2);
+
+        testSamePullRequestsTriggersMultipleProjects(project2);
+
+    }
+
+    public void testAddingProjectToGHRepository(FreeStyleProject project2) throws Exception {
+
+        LOGGER.fine("test ---- testAddingProjectToGHRepository\n");
+
+        // Check that job 2 has no builds
+
+        new Condition() {
+            public boolean satisfied() {
+                return project2.getLastBuild() != null;
+            }
+        }.waitUntilSatisfied(NEXTBUILD_TIMEOUT);
+
+        Assert.assertNull(MessageFormat.format("Build in job 2 should not be triggered yet, on opening of pull request {0} after {1} minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), TimeUnit.SECONDS.toMinutes(NEXTBUILD_TIMEOUT)),
+                project2.getLastBuild() );
+
+        // Wait till existing trigger Phrase is built, then send new trigger comment to both projects
+
+        final List<JSONObject> instances = new ArrayList<JSONObject>();
+
+        int count = 0;
+        LOGGER.fine("Testing trigger Phrase...");
+        final String triggerPhrase = "Jenkins test this please ";
+        pullRequest.comment(triggerPhrase + (count++) );
+        Assert.assertNull(MessageFormat.format("Unexpected build triggered with comment ''{0}''",
+                triggerPhrase),
+                waitForNextBuild(NEXTBUILD_TIMEOUT, "Waiting build triggered with comment " + triggerPhrase));
+
+        updateTriggerPhrase(triggerPhrase);
```

Comment:
The use of this method is confusing. I would change the signature to need always a project.

Reply:
If you use an IDE, that is not a problem. From the particular method you call the general one and inline the old method.